crates/storage/provider/src/providers/blockchain_provider.rs

```diff
@@ -1458,36 +1462,51 @@ mod tests {
     };
     use reth_execution_types::{Chain, ExecutionOutcome};
     use reth_primitives::{
-        BlockHashOrNumber, BlockNumHash, BlockNumberOrTag, BlockWithSenders, Header, Receipt,
-        SealedBlock, SealedBlockWithSenders, StaticFileSegment, Withdrawals, B256,
+        BlockHashOrNumber, BlockNumHash, BlockNumberOrTag, BlockWithSenders, Receipt, SealedBlock,
+        SealedBlockWithSenders, StaticFileSegment, TransactionMeta, TransactionSignedNoHash,
+        Withdrawals, B256,
     };
     use reth_storage_api::{
         BlockHashReader, BlockIdReader, BlockNumReader, BlockReader, BlockReaderIdExt, BlockSource,
         ChangeSetReader, HeaderProvider, ReceiptProviderIdExt, RequestsProvider,
-        TransactionVariant, WithdrawalsProvider,
+        TransactionVariant, TransactionsProvider, WithdrawalsProvider,
     };
     use reth_testing_utils::generators::{
         self, random_block, random_block_range, random_changeset_range, random_eoa_accounts,
-        random_receipt, random_signed_tx, BlockParams, BlockRangeParams,
+        random_receipt, BlockParams, BlockRangeParams,
     };
     use revm::db::BundleState;
+    use std::ops::Bound;
 
     const TEST_BLOCKS_COUNT: usize = 5;
 
+    const TEST_TRANSACTIONS_COUNT: u8 = 4;
+
     fn random_blocks(
         rng: &mut impl Rng,
         database_blocks: usize,
         in_memory_blocks: usize,
         requests_count: Option<Range<u8>>,
         withdrawals_count: Option<Range<u8>>,
+        tx_count: impl RangeBounds<u8>,
     ) -> (Vec<SealedBlock>, Vec<SealedBlock>) {
         let block_range = (database_blocks + in_memory_blocks - 1) as u64;
+
+        let tx_start = match tx_count.start_bound() {
+            Bound::Included(&n) | Bound::Excluded(&n) => n,
+            Bound::Unbounded => u8::MIN,
+        };
+        let tx_end = match tx_count.end_bound() {
+            Bound::Included(&n) | Bound::Excluded(&n) => n + 1,
```

Comment:
included and excluded bounds need to be handled separately, see https://github.com/paradigmxyz/reth/blob/019ec727a2e078fd1517d5b04f39f6713d3bf0d7/crates/storage/provider/src/lib.rs#L49-L63

Reply:
I think in this case it doesn't work because if as an input I've `0..0`, then:
- start_bound is included so that start = 0
- end_bound is excluded so that end = 0

And we end up with `0..0` leading to the error `UniformSampler::sample_single: low >= high` in the `random_block_range` method. I think that for testing we can keep this logic, adding 1 on the output terminal seems enough to do what we want, right?

Reply:
And in `BlockRangeParams` we should pass a range and not a range inclusive... Maybe I missed something, sorry :)

Reply:
ah I see, why can't we just pass `TEST_TRANSACTIONS_COUNT..=TEST_TRANSACTIONS_COUNT` then? IMO `0..0` doesn't make sense because, as you described, it's an empty range.

Reply:
sorry, took me a while to figure that out... Checked out the branch myself and saw it, yep, agreed. Let's leave it as-is now and improve in a separate PR.